src/sonic_ax_impl/mibs/ietf/rfc1213.py

```diff
@@ -54,32 +54,64 @@ class IfTypes(int, Enum):
 class ArpUpdater(MIBUpdater):
     def __init__(self):
         super().__init__()
+        self.db_conn = Namespace.init_namespace_dbs()
         self.arp_dest_map = {}
         self.arp_dest_list = []
         self.arp_dest_map = {}
         self.arp_dest_list = []
+        self.neigh_key_list = {}
 
-    def update_data(self):
-        self.arp_dest_map = {}
-        self.arp_dest_list = []
+    def reinit_data(self):
+        Namespace.connect_all_dbs(self.db_conn, mibs.APPL_DB)
+        self.neigh_key_list = Namespace.dbs_keys_namespace(self.db_conn, mibs.APPL_DB, "NEIGH_TABLE:*")
+
+    def _update_host_data(self):
         for entry in python_arptable.get_arp_table():
             dev = entry['Device']
             mac = entry['HW address']
             ip = entry['IP address']
+            self._update_arp_info(dev, mac, ip)
+
+    def _update_namespace_data(self):
+        for neigh_key in self.neigh_key_list:
+            neigh_str = neigh_key.decode()
+            db_index = self.neigh_key_list[neigh_key]
+            neigh_info = self.db_conn[db_index].get_all(mibs.APPL_DB, neigh_key, blocking=True)
```

Comment:
>blocking=True [](start = 81, length = 13)

Let's use non blocking, which make snmpagent robust even if there is an issue in any namespace, such as process crash, DB corruption.

Reply:
Updated as per review comment.

Reply:
Non blocking may return a None (please verify), then you need to handle this corner case.

---
In reply to: [458323712](https://github.com/Azure/sonic-snmpagent/pull/129#discussion_r458323712) [](ancestors = 458323712)

Reply:
Added check, will continue with other keys if None is returned on get_all on any key.